controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -439,6 +459,9 @@ func (r *KubeadmControlPlaneReconciler) generateMachine(ctx context.Context, kcp
 			},
 		},
 	}
+	if fd != "" {
+		machine.Spec.FailureDomain = &fd
```

Comment:
Why not return `*string` from the function so we can just assign it to the Spec?

Reply:
What's the benefit of return `*string` over how I've done it?

Reply:
We can assign it to Spec without the check if `fd` is an empty string